server/registry/internal/storage/logger.go

```diff
@@ -58,7 +58,7 @@ func (l gormLogger) Trace(ctx context.Context, begin time.Time, fc func() (sql s
 		"duration": time.Since(begin),
 	})
 
-	if err != nil && err != gorm.ErrRecordNotFound {
+	if err != nil && err != gorm.ErrRecordNotFound && !alreadyExists(err) {
```

Comment:
A larger question than this PR, but should we consider using [errors.Is](https://pkg.go.dev/errors#Is)?

Reply:
Sure, I think that would be fine. In this situation the `alreadyExists` function requires looking at status codes attached to the errors so I think it might not apply.